hig.js

```diff
@@ -87,17 +88,35 @@ function speak(phrases) {
 		return false;  	
 	}
 	var p = phrases.shift();
-	var v = 'Alex';
-	if (p.name == ANNOUNCER) {
-		v = 'Samantha';
-	} 
+	if(os == 'darwin'){  //mac male voice to talk as hockey player
+		var v = 'Alex';
+			if (p.name == ANNOUNCER ) { //mac female voice to talk as announcer 
+				v = 'Samantha';
+		} 
+	}
+	else if (os=='win32'){ //windows male voice to talk as hockey player
+		 var v = 'David';
+			 if (p.name == ANNOUNCER ) { //windows female voice to talk as announcer 
+				 v = 'Zira';
+		}
+	}
 	var ph = p.phrase.toString().startsWith('(') ? "..." : p.phrase;
-	say.speak(ph, v, 1, (err) => {
-  		if (err) {
-    		return console.error(err);
-  		}
-  		speak(phrases);
-	});		
+	let p1 = new Promise(
+        // The executor function is called with the ability to resolve or
+        // reject the promise
+       (resolve, reject) => {
+			setTimeout( function() {
```

Comment:
There is something wrong with the timeout, now. When I run this from your fork there is a huge lag between statements so something is wrong here. does it work properly for you? There should not be a huge lag between statements, it should flow naturally like a normal conversation. Is this what you hear on windows? I think it might have something to do  with your timeout value, what is the function of that?

Reply:
I just added some timeout experimentation. It seems to work pretty well but there could probably be a more dynamic approach by maybe adding keys to the json file which report how long of a timeout is necessary for each phrase. For example, some of the questions and statements are longer than others so the flow cannot be perfect by having a universal timeout for each phrase type.  It would be nice to have a timeout key which can be referenced and maybe passed to the next function call. So if phrase x requires a 3 second timeout it sends "3000" to the next call letting that phrase know it must wait 3000 milliseconds before it starts. This would allow me to delete the conditionals I just added and simply have one call with an ever changing timeout parameter.

Also, I will research other ways of doing this but right now I am unaware of any way of knowing when the speaker has finished talking except by measuring.

Reply:
Nevermind, I got the windows voices to work properly and have a better handle on the timing. I might try a different strategy.

Reply:
Now that it is configured correctly, the promise is working as it should but there is still a delay between phrases which I am working on now.

Reply:
You can use promises to handle the situation i described above. Even though my promise was written correctly, because my windows voice was not configured correctly it was not behaving properly. Now that it is, I can use the promise code to make sure that the previous phrase has been completed.  The problem now is the delay is still too long.

Reply:
The delay is too long because it is synchronous. The phrase completes but there is not a natural flow between phrases.  The only way to get a natural flow from what I can tell is to use asynchronous timeouts and measure the timeouts.  There does not seem to be a way around measuring how long each phrase will take to complete. Perhaps there can be some sort of script which adds the proper timeout time for you.

Reply:
I have an attempt at trying to time it up using the asynchronous model but it is not good enough yet. The synchronous way works at least and the delay is better than it was this morning. Ive been trying to figure out how to time these phrases up by research, planning and minor experimentation because I have some ideas that require editing the json file and such and I do not want to do a ton of work if there is an easier way to accomplish it.

Reply:
The most standard way to "promisify" the JS code is to essentially build off of promise-based building blocks.
Right now, `say.speak()` is a classic callback-driven node.js function.
You can try to promisify it, via node.js `Util.promisify()`. 

That will give you a function "promiseToSpeak()" or something like that.

Then your flow should be:
promiseToSpeak("something")
  .then(() => promiseToSpeak("something else"))
  .then(()=> promiseToSpeak("and more and more))
  .then(() => ..... whatever it is to do then....)
  .catch(err => ... error handling....)

Reply:
The timeouts make zero sense.

Reply:
LOL, thanks Timotej. I very much appreciate it. Yeah, my local code is different as I was playing around with it last night. I started to understand how this works BUT I have not figured out how to have the next promise speak when the narrator has stopped speaking BECAUSE the promise resolves before the speaker finishes talking when using the asynchronous model.

Reply:
Also, thank you so much. This is great advice and a clear way of explaining it.